lib/id3c/cli/command/etl/fhir.py

```diff
@@ -469,6 +472,41 @@ def process_encounter(db: DatabaseSession, encounter: Encounter,
         details         = details)
 
 
+def process_encounter_source(encounter: Encounter) -> Optional[Dict[str, str]]:
+    """
+    Returns the source of the given *encounter*. Splits the source into
+    individual keys from a REDCap URL with the following format:
+
+    https://redcap.iths.org/<project_id>/<record_id>/<event_name>/<repeat_instance>
+
+    Note: <event_name> and <repeat_instance> are only included if the
+    REDCap project has repeating events.
+    """
+    if not encounter.meta:
+        return None
+
+    encounter_source = encounter.meta.source
+
+    if not encounter_source:
+        return None
+
+    # Used named group pattern for easy conversion to dictionary after matching
+    # Include letters a-f in `redcap_record_id` for the barcode record id in UW Retros
+    pattern = re.compile(r'https://redcap\.iths\.org' + \
+                         r'(?P<redcap_project_id>/[0-9]+)' + \
+                         r'(?P<redcap_record_id>/[a-z0-9]+)' + \
```

Comment:
Reading your comment above, is this supposed to say `[a-f]` instead of `[a-z]`?

Reply:
Yes! Thank you for catching that!